### PR TITLE
ADEN-7362 Prepare AE for outstream template

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: src/listeners/slot-listener.js [hunk 1]
```diff
@@ -88,6 +88,8 @@ class SlotListener {
 			case 'collapse':
 				adSlot.collapse();
 				break;
+			case 'manual':
```
Comment (jacekbj) [inline src/listeners/slot-listener.js]: Do we have a list of "ad types"?

File: src/utils/dimensions.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ export function getViewportHeight() {
 	return Math.max(document.documentElement.clientHeight, window.innerHeight || 0);
 }
 
-export function isInViewport(element, topOffset = 0, bottomOffset = 0) {
+export function isInViewport(element, topOffset = 0, bottomOffset = 0, areaThreshold = 0.5) {
```
Comment (jacekbj) [inline src/utils/dimensions.js]: Please add jsdoc.

File: src/video/player/porvata/porvata.js [hunk 3]
```diff
@@ -344,4 +346,16 @@ export class Porvata {
...
 			});
 	}
+
+	static isVpaid(contentType) {
+		return contentType === 'application/javascript';
+	}
+
+	static isVideoAutoplaySupported() {
+		const isAndroid = client.getOperatingSystem() === 'Android';
+		const browserInfo = client.getBrowser().split(' ');
```
Comment (jacekbj) [inline src/video/player/porvata/porvata.js]: It's funny that 2 last lines of `getBrowser()` method are:
```
browser = matches.join(' ');

return browser;
```

File: src/video/player/porvata/porvata.js [hunk 4]
```diff
@@ -344,4 +346,16 @@ export class Porvata {
...
+		const isCompatibleChrome = browserInfo[0].indexOf('Chrome') !== -1 && parseInt(browserInfo[1], 10) >= 54;
```
Comment (jacekbj) [inline src/video/player/porvata/porvata.js]: https://github.com/video-dev/can-autoplay funny how it tries to insert video/audio element into website and play it :D I couldn't find "source of truth" for supported browsers so I think this is fine.

File: src/listeners/slot-listener.js (hunk 1 above)
Comment (fraszczakszymon) [reply src/listeners/slot-listener.js]: There are a couple of them in the running DFP creatives. But ad-engine isn't going to handle all of them. Here is the main list.

File: src/video/player/porvata/porvata.js (hunk 3 above)
Comment (fraszczakszymon) [reply src/video/player/porvata/porvata.js]: You're right. It doesn't makes sense. Fixed :)

File: src/utils/dimensions.js (hunk 2 above)
Comment (fraszczakszymon) [reply src/utils/dimensions.js]: Added

File: src/video/player/porvata/porvata.js (hunk 4 above)
Comment (fraszczakszymon) [reply src/video/player/porvata/porvata.js]: We can consider that. But I would run it in separate story/epic. We would need to check what's the overall impact

**Pull request merged**